lib/classes/board.rb

```diff
@@ -0,0 +1,77 @@
+module OurTicTacToe
+	class Board
+		attr_reader :grid
+		def initialize(grid_board = {})
+			@grid = grid_board.fetch(:grid, default_board)
+		end
+
+		def get_cell(row, col)
+			grid[row][col]
+		end
+
+		def set_cell(row, col, value)
+			cell = get_cell(row, col)
+			cell.value.empty? ? cell.value = value : nil
+		end
+
+		def game_over
+			return :winner if winner?
+			return :draw if draw?
+
+			false
+		end
+
+		def formatted_grid
+			grid.each_with_index do |row, outer|
+				r = row.each_with_index.map do |col, inner|
+					if col.value.empty?
+						case outer
+						when 0
+							inner + 1
+						when 1
+							inner + 4
+						when 2
+							inner + 7
+						end
+					else
+						col.value
+					end
+				end
+				puts r.join(' ')
+			end
+		end
+
+		def winning_positions
+			grid + grid.transpose + diagonals
+		end
+
+		def diagonals
+			[
+				[get_cell(0, 0), get_cell(1, 1), get_cell(2, 2)],
+				[get_cell(0, 2), get_cell(1, 1), get_cell(2, 0)]
+			]
+		end
```

Comment:
I try to avoid hardcoded values when possible. Can we generate the combinations for the diagonals?

Reply:
Created an issue #15  out of it to solve at later date. We really appreciate the observation!